File: problem2.py
'''
Tightly Packed
'''
import math

def __main__():
  n = long(raw_input())
  minDiff = float("inf")

  x = long(math.ceil(math.sqrt(n)))
  while True:
    y = (math.ceil(n / float(x)))
    if x > 2*y: break

    diff = x*long(y) - n
    if diff < minDiff and diff >= 0:
      minDiff = int(diff)
      if minDiff is 0: break

    x += 1

  print minDiff

__main__()

File: problem2.cpp
/*
Tightly Packed
*/
#include <iostream>
#include <cmath>
#include <limits>

using namespace std;

int main() {
  long N; cin >> N;

  long x = ceil(sqrt(N));
  long minDiff = numeric_limits<long>::max();

  while(true) {
    long y = ceil(N / (double) x);
    if(x > 2*y) break;

    int diff = x * y - N;
    if (diff < minDiff && diff >= 0) {
      minDiff = diff;
      if (minDiff == 0) break;
    }

    x++;
  }

  cout << minDiff << endl;
}
